UI - Fix intl.translate in react components

carlito-scality opened the pull request:
**Component**:UI

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

- Once IntlGlobalProvider is introduced to allow us to translate text in sagas, intl.translate does not work anymore in React components, only intl.messages works. It triggers an JS error 'translate is not a function'. We need to use ONLY IntlGlobalProvider's intl.translate everywhere (saga and react components) for translation to be consistent. Thus we can get rid of HOC injectIntl

**Summary**:

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: https://github.com/scality/metalk8s/issues/2122

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in ClusterMonitoring
- we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

bert-e commented:
# Hello carlito-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




There is a conflict between your branch `bugfix/fix-intl-translate-in-react-components` and the
destination branch `development/2.4`.

Please resolve the conflict on **the feature branch** (`bugfix/fix-intl-translate-in-react-components`).

```bash
 $ git fetch
 $ git checkout origin/bugfix/fix-intl-translate-in-react-components
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:bugfix/fix-intl-translate-in-react-components
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

slaperche-scality commented on ui/src/translations/IntlGlobalProvider.js:
```diff
@@ -36,10 +36,13 @@ class IntlTranslator {
   // Formatting Functions
   // ------------------------------------
   translate(key, values) {
+    if (!key) {
+      return '';
+    }
```
this is the only relevant change, the rest of the diff is linting/cosmetics and belong to a dedicated commit

carlito-scality committed:
UI: I18n - intl.translate should return empty string when key is falsy

bert-e commented:
# Branches have diverged



This pull request's source branch `bugfix/fix-intl-translate-in-react-components` has diverged from
`development/2.5` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.5` into `bugfix/fix-intl-translate-in-react-components`
* Rebase `bugfix/fix-intl-translate-in-react-components` onto `origin/development/2.5`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

carlito-scality commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Branches have diverged



This pull request's source branch `bugfix/fix-intl-translate-in-react-components` has diverged from
`development/2.5` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.5` into `bugfix/fix-intl-translate-in-react-components`
* Rebase `bugfix/fix-intl-translate-in-react-components` onto `origin/development/2.5`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in About
    - we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in CreateVolume
    - we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in EnvironmentCreationForm
        - we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in Layout
        - we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in LoadingCallback
        - we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in NodeCreateForm
- we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in NodeDeployment
- we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in NodeInformation
- we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in NodeList
- we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in NodeVolumes
- we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in SolutionList
- we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

carlito-scality committed:
UI: replace injectIntl's intl.messages by IntlGlobalProvider's intl.translate in VolumeInformation
- we need to use IntlGlobalProvider's intl.translate everywhere (saga and react components) to be consistent

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

ChengYanJin reviewed (approved):
LGTM!

carlito-scality commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

carlito-scality commented:
/wait

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



Please check the status of the associated issue None.

Goodbye carlito-scality.
